fastlane/Fastfile

```diff
@@ -866,49 +865,85 @@ platform :ios do
+      match(readonly: true, platform: "ios")
       increment_build_number(
         build_number: latest_testflight_build_number(platform: 'ios') + 1,
-        xcodeproj: 'PurchaseTester.xcodeproj'
+        xcodeproj: xcodeproj_path,
       )
+    end
+    
+    build_ios_app(
+      workspace: 'RevenueCat.xcworkspace',
+      scheme: "PurchaseTester",
+      destination: "generic/platform=ios",
+      xcodebuild_formatter: '',
+      export_method: 'app-store',
+      export_options: {
+        manageAppVersionAndBuildNumber: true,
+        provisioningProfiles: {
+          "com.revenuecat.sampleapp" => "match AppStore com.revenuecat.sampleapp",
+          "com.revenuecat.sampleapp.watchkitapp" => "match AppStore com.revenuecat.sampleapp.watchkitapp"
+        }
+      }
+    )
 
-      match(readonly: true)
-      build_ios_app(
-        workspace: "PurchaseTester.xcworkspace",
-        scheme: "PurchaseTester",
-        destination: "generic/platform=ios",
-        xcodebuild_formatter: ''
-      )
+    if not dry_run
       upload_to_testflight(
-        changelog: changelog
-      )
+      changelog: changelog,
+      app_identifier: "com.revenuecat.sampleapp",
+    )
+    else
+      UI.message("Dry run mode enabled. Skipping upload to TestFlight")
     end
 
-    def macos_build(changelog)
+    # The following commands use configuration that lives in the PurchaseTesterSwiftUI directory, 
+    # like the Matchfile. So we move to that project's directory to run them.
+    Dir.chdir("#{purchase_tester_root}/fastlane") do
+      match(readonly: true, platform: "macos", additional_cert_types: "mac_installer_distribution")
       increment_build_number(
         build_number: latest_testflight_build_number(platform: 'osx') + 1,
-        xcodeproj: 'PurchaseTester.xcodeproj'
+        xcodeproj: xcodeproj_path,
       )
+    end
 
-      match(readonly: true, platform: "macos", additional_cert_types: "mac_installer_distribution")
-      build_mac_app(
-        workspace: "PurchaseTester.xcworkspace",
-        scheme: "PurchaseTester",
-        destination: 'generic/platform=macOS',
-        xcodebuild_formatter: ''
-      )
+    build_mac_app(
+      workspace: 'RevenueCat.xcworkspace',
+      scheme: "PurchaseTester",
+      destination: 'generic/platform=macOS',
+      xcodebuild_formatter: '',
+      export_method: 'app-store',
+      export_options: {
+        manageAppVersionAndBuildNumber: true,
+        provisioningProfiles: {
+          "com.revenuecat.sampleapp" => "match AppStore com.revenuecat.sampleapp macos",
+        }
+      }
+    )
+
+    if not dry_run
       upload_to_testflight(
         changelog: changelog,
-        app_platform: 'osx'
+        app_platform: 'osx',
+        app_identifier: "com.revenuecat.sampleapp",
       )
+    else
+        UI.message("Dry run mode enabled. Skipping upload to TestFlight")
     end
-
-    ios_build(changelog)
-    macos_build(changelog)
```

Comment:
I also removed the local functions because I wanted to reuse a couple of the parameters without having to pass them in over and over. I can re-add them, but I don't usually find local functions to be easier to read. Thoughts?

Reply:
I'm ok with that. I think it only could make some sense in some situations where the code needs to be executed multiple times, so I agree with the change 👍